posix/mount.c

```diff
@@ -0,0 +1,286 @@
+
+static void _free_mount_table(void)
+{
+    for (size_t i = 0; i < _mount_table_size; i++)
+        oe_free(_mount_table[i].path);
+}
+
+oe_device_t* oe_mount_resolve(const char* path, char suffix[OE_PATH_MAX])
+{
+    oe_device_t* ret = NULL;
+    size_t match_len = 0;
+    char realpath[OE_PATH_MAX];
+    bool locked = false;
+
+    if (!path || !suffix)
+        OE_RAISE_ERRNO(OE_EINVAL);
+
+    /* First check whether a device id is set for this thread. */
+    {
+        uint64_t devid;
+
+        if ((devid = oe_get_thread_devid()) != OE_DEVID_NONE)
+        {
+            oe_device_t* device;
+
+            if (!(device =
+                      oe_device_table_get(devid, OE_DEVICE_TYPE_FILE_SYSTEM)))
+                OE_RAISE_ERRNO(OE_EINVAL);
+
+            /* Use this device. */
+            if (oe_strlcpy(suffix, path, OE_PATH_MAX) >= OE_PATH_MAX)
+                OE_RAISE_ERRNO(OE_ENAMETOOLONG);
+
+            ret = device;
+            goto done;
+        }
+    }
+
+    /* Find the real path (the absolute non-relative path). */
+    if (!oe_realpath(path, realpath))
+        OE_RAISE_ERRNO(oe_errno);
+
+    oe_conditional_lock(&_lock, &locked);
+
+    /* Find the longest binding point that contains this path. */
+    for (size_t i = 0; i < _mount_table_size; i++)
+    {
+        size_t len = oe_strlen(_mount_table[i].path);
+        const char* mpath = _mount_table[i].path;
+
+        if (mpath[0] == '/' && mpath[1] == '\0')
+        {
+            if (len > match_len)
+            {
+                if (suffix)
+                {
+                    oe_strlcpy(suffix, realpath, OE_PATH_MAX);
+                }
+
+                match_len = len;
+                ret = _mount_table[i].fs;
+            }
+        }
+        else if (
+            oe_strncmp(mpath, realpath, len) == 0 &&
+            (realpath[len] == '/' || realpath[len] == '\0'))
+        {
+            if (len > match_len)
+            {
+                if (suffix)
+                {
+                    oe_strlcpy(suffix, realpath + len, OE_PATH_MAX);
+
+                    if (*suffix == '\0')
+                        oe_strlcpy(suffix, "/", OE_PATH_MAX);
+                }
+
+                match_len = len;
+                ret = _mount_table[i].fs;
+            }
+        }
+    }
+
+    if (!ret)
+        OE_RAISE_ERRNO_MSG(OE_ENOENT, "path=%s", path);
+
+done:
+    oe_conditional_unlock(&_lock, &locked);
+
+    return ret;
+}
+
+int oe_mount(
+    const char* source,
+    const char* target,
+    const char* filesystemtype,
+    unsigned long mountflags,
+    const void* data)
+{
+    int ret = -1;
+    oe_device_t* device = NULL;
+    oe_device_t* new_device = NULL;
+    bool locked = false;
+
+    OE_UNUSED(data);
+
+    if (!target || !filesystemtype)
+        OE_RAISE_ERRNO(OE_EINVAL);
+
+    /* Resolve the device for the given filesystemtype. */
+    if (!(device =
+              oe_device_table_find(filesystemtype, OE_DEVICE_TYPE_FILE_SYSTEM)))
```

Comment:
It would not. The **mount()** function is entirely an enclave-side phenomenon. One could implement an "ext4" file system though within the encalve. The **mount()** call, unlike many ohter calls, is never forwarded to the host.